userspace/libsinsp/filterchecks.cpp

```diff
@@ -4832,23 +4833,7 @@ uint8_t* sinsp_filter_check_group::extract(sinsp_evt *evt, OUT uint32_t* len, bo
 			unordered_map<uint32_t, scap_groupinfo*>::iterator it;
 
 			ASSERT(m_inspector != NULL);
-			unordered_map<uint32_t, scap_groupinfo*>* grouplist =
-				(unordered_map<uint32_t, scap_groupinfo*>*)m_inspector->get_grouplist();
-			ASSERT(grouplist->size() != 0);
-
-			if(tinfo->m_gid == 0xffffffff)
-			{
-				return NULL;
-			}
-
-			it = grouplist->find(tinfo->m_gid);
-			if(it == grouplist->end())
-			{
-				ASSERT(false);
-				return NULL;
-			}
-
-			scap_groupinfo* ginfo = it->second;
+			auto ginfo = m_inspector->get_group(tinfo->m_gid);
```

Comment:
It's completely legit if a gid doesn't map to a group name (even if our nss cache was up to date) so I don't think we should ASSERT below but just treat the group name as empty (or something like this, maybe "<NA>"?)

A nullptr is doubly valid since you do the gid == -1 check inside the method